src/QueryComplexity.ts

```diff
@@ -190,6 +190,10 @@ export default class QueryComplexity {
               if (!field) {
                 break;
               }
+              // Prevent astNode's value is undefined when perform `_entities` & `_service` query
+              if (!field.astNode) {
+                (field.astNode as any) = {}
+              }
```

Comment:
I don't like that we are mutating the query object here in the validation rule. Could you provide a test case that fails or some more context when this is needed? I don't quite understand the need for this yet...

Reply:
@ivome Thanks for feedback. When I migrated schema stitching in my app to Apollo Federation. When the Apollo Gateway starts, it will send `_service { sdl }` query to federation services. And after the query is executed, `getComplexity` (with directiveEstimator enabled) in will throw an error. I will provide the test case for above cases.

Reply:
Ah, I see now. That's probably because that field was not created via GraphQL SDL and therefore has no AST. But I would fix that in the directive estimator and check for the existence of the `astNode` on the field. That way we don't have to mutate the query object...

Reply:
@ivome Thanks. I've just updated my PR. Could you please check?